imap/imapd.c

```diff
@@ -3828,6 +3828,8 @@ static void cmd_append(char *tag, char *name, const char *cur_name)
     const char *parseerr = NULL, *url = NULL;
     struct appendstage *curstage;
     mbentry_t *mbentry = NULL;
+    size_t maxsize = config_getint(IMAPOPT_APPEND_MAXSIZE) * 1024;
+    if (!maxsize) maxsize = UINT32_MAX;
 
```

Comment:
Would the call to config_getint be reduced to just one for the whole program lifetime, if maxsize is declared `static`? (as opposed to executing config_getint anytime cmd_append is called)
```c
static size_t maxsize;
if (!maxsize) {
  maxsize = config_getint(IMAPOPT_APPEND_MAXSIZE) * 1024;
  if (!maxsize) maxsize = UINT32_MAX;
}

Reply:
The cost of cmd_append is so high that this kind of optimisation isn't worth making the code more complex for.  Optimisations in tight loops are great, but optimisations at this level aren't worth it.